packages/server/tests/acceptance/ws/subscribe.spec.ts

```diff
@@ -20,39 +20,120 @@
+        assert.equal(expectedNumberOfOpenConnections, true, `Incorrect number of open connections. Expected ${process.env.CONNECTION_LIMIT}, got ${server._connections}`);
+
+        process.removeAllListeners('uncaughtException');
+        listeners.forEach(async (listener) => {
+            process.on('uncaughtException', listener);
+
+        });
+    });
+
+    try {
+        process.nextTick(async () => {
+            await establishConnection();
+        });
+    } catch (err) {
+        console.log('Caught!', err.message);
+    }
+};
+
+describe('@web-socket Acceptance Tests', async function() {
     this.timeout(240 * 1000); // 240 seconds
     const CHAIN_ID = process.env.CHAIN_ID || 0;
+    let server;
 
     // cached entities
     let requestId;
     let wsProvider;
 
-    this.beforeAll(async () => {
-        wsProvider = new ethers.providers.WebSocketProvider(
+    this.beforeEach(async () => {
+        const { socketServer } = global;
+        server = socketServer;
+        
+        wsProvider = await new ethers.providers.WebSocketProvider(
             `ws://localhost:${process.env.WEB_SOCKET_PORT}`
         );
-    });
 
-    this.beforeEach(async () => {
         requestId = Utils.generateRequestId();
+        // Stabilizes the initial connection test.
+        await new Promise(resolve => setTimeout(resolve, 1000));
+    });
+
+    this.afterEach(async () => {
+        wsProvider.destroy();
     });
 
-    describe('Connection', async function () {
-        it('establishes connection', async function () {
+
+    describe('Connection', async function() {
+        it('establishes connection', async function() {
             expect(wsProvider).to.exist;
             expect(wsProvider._wsReady).to.eq(true);
         });
 
-        it('Socket server responds to the eth_chainId event', async function () {
+        it('Socket server responds to the eth_chainId event', async function() {
             const response = await wsProvider.send('eth_chainId');
             expect(response).to.eq(CHAIN_ID);
         });
+
+        it('Establishes multiple connections', async function() {
+
+            const secondProvider = new ethers.providers.WebSocketProvider(
+                    `ws://localhost:${process.env.WEB_SOCKET_PORT}`
+            );
+
+            const response = await secondProvider.send('eth_chainId');
+            expect(response).to.eq(CHAIN_ID);
+            expect(server._connections).to.equal(2);
+
+            secondProvider.destroy();
+        });
+
+        it('Does not allow more connections than the connection limit', async function() {
+            // We already have one connection
+            for (let i = 1; i < parseInt(process.env.CONNECTION_LIMIT); i++) {
+                await establishConnection();
+            }
+
+            expect(server._connections).to.equal(parseInt(process.env.CONNECTION_LIMIT));
```

Comment:
Can we add a test goes through checking 0, n then 0 connections in a single session to make sure the increment and decrement logic in a single session is good when connection are added and closed.

Reply:
That counter decrement line "connectedClients--;" is confusing the conversation.  Apologies.  I thought I removed it earlier.  The number of connections is now simply read from the server context on line 65, 
```connectedClients = ctx.app.server._connections;``` and then checked in the verifyClient on line 49:
```
if (connectedClients >= MAX_CONNECTIONS) {
                return done(false, 429, 'Connection limit exceeded');
} 
```
If the limit has been reached, then the next connection is rejected on verifyClient before it even reached the decrement of the counter.  The tests pass because the limit breaking connection is immediately rejected and closed.